system/lib/libc/lookup_name.c

```diff
@@ -0,0 +1,27 @@
+// Emscripten-specific version of musl/src/network/lookup_name.c
+
+#include <string.h>
+#include "musl/src/network/lookup.h"
+#include "emscripten_internal.h"
+
+int __lookup_name(struct address buf[static MAXADDRS], char canon[static 256], const char *name, int family, int flags)
+{
+	if (family != AF_INET || flags != AI_CANONNAME) {
```

Comment:
I don't see `AI_CANONNAME` in the old impl - is that needed because the new musl code used will pass it, or something like that?

Reply:
We are implementing a slightly different function here.  This `__lookup_name` is lower level than the old `gethostbyname` so you shouldn't be looking to necessarily compare them exactly.

The `AI_CANONNAME` is justa check to make sure that we don't have any other callers other than this one: https://github.com/emscripten-core/emscripten/blob/d01d57e0b432ab209021bd47380d87bb1f8b647c/system/lib/libc/musl/src/network/gethostbyname2_r.c#L21

I will change this to an assert and add a comment to make this more clear.